runtime/Go/antlr/tree.go

```diff
@@ -101,10 +94,10 @@ type BaseParseTreeListener struct{}
 
 var _ ParseTreeListener = &BaseParseTreeListener{}
 
-func (l *BaseParseTreeListener) VisitTerminal(node TerminalNode)      {}
```

Comment:
I understand the impetus to change the listener implementation, but I think that a more focused pull request (just adding support for Visitor) would be better here.  Especially if the Listener change is going to be a breaking one.

The Visitor never worked, so implementing it should be straightforward, but breaking the way Listener worked (and the unrelated changes) seem like they'd be better as a separate pull request.